base_layer/core/src/proof_of_work/monero_rx.rs

```diff
@@ -231,25 +231,343 @@ fn verify_root(monero_data: &MoneroData) -> Result<(), MergeMineError> {
     Ok(())
 }
 
-fn merged_mining_subfield(header: &BlockHeader) -> SubField {
-    let hash = header.merged_mining_hash();
-    let depth = 0;
-    SubField::MergeMining(VarInt(depth), Hash::from(from_slice(&hash)))
-}
-
 fn verify_header(header: &BlockHeader, monero_data: &MoneroData) -> Result<(), MergeMineError> {
-    if !(monero_data
-        .coinbase_tx
-        .prefix
-        .extra
-        .0
-        .contains(&merged_mining_subfield(header)))
-    {
+    let mut found = false;
+    for item in monero_data.coinbase_tx.prefix.extra.0.clone() {
+        match item {
+            SubField::MergeMining(x, y) => {
+                let compare = Hash::from(from_slice(&header.merged_mining_hash()));
+                if (x, y) == (VarInt(0), compare) {
+                    found = true;
```

Comment:
Can we exit early if the field is found - or is it important to fail if there are any other MergeMining fields which don't match?

Reply:
Thanks, should exit as soon as a match occurs.

Reply:
In that case I suggest using the `Iterator::any` call

```rust
let monero_data = &monero_data.coinbase_tx.prefix.extra.0;
let is_found = monero_data.iter().any(|item| {
        match item {
            SubField::MergeMining(x, y) => 
depth == VarInt(0) && merge_mining_hash == hash,
_ => false 
}});
```